pkg/api/configsync/v1beta1/rootsync_types.go

```diff
@@ -58,8 +59,9 @@ type RootSyncSpec struct {
 	// Must be one of git, oci, helm. Optional. Set to git if not specified.
 	// +kubebuilder:validation:Pattern=^(git|oci|helm)$
```

Comment:
I think it might make sense to change this to an `enum` instead of a `pattern`, but this results in a CRD diff so I think that should be a separate change

Reply:
Agreed. We can change `sourceFormat` to enum as well.